ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -139,7 +139,12 @@ private[ember] object Encoder {
           }
         }
 
-        if (!chunked && !appliedContentLength && !NoPayloadMethods.contains(req.method)) {
+        if (
+          !appliedContentLength && req.body == EmptyBody && !NoPayloadMethods.contains(req.method)
```

Comment:
should one maybe use reference equality `eq` directly instead of `equals(==)` ?

Reply:
Good point, I'll make this change in https://github.com/http4s/http4s/pull/6753. We need to fix it in at least one other place.